fdbserver/include/fdbserver/DataDistribution.actor.h

```diff
@@ -349,6 +351,145 @@ struct ShardTrackedData {
 	Reference<AsyncVar<Optional<ShardMetrics>>> stats;
 };
 
+// Holds the permitted size and IO Bounds for a shard
+struct ShardSizeBounds {
+	StorageMetrics max;
+	StorageMetrics min;
+	StorageMetrics permittedError;
+
+	bool operator==(ShardSizeBounds const& rhs) const {
+		return max == rhs.max && min == rhs.min && permittedError == rhs.permittedError;
+	}
+};
+
+// Gets the permitted size and IO bounds for a shard
+ShardSizeBounds getShardSizeBounds(KeyRangeRef shard, int64_t maxShardSize);
+
+// Determines the maximum shard size based on the size of the database
+int64_t getMaxShardSize(double dbSizeEstimate);
+
+#ifndef __INTEL_COMPILER
+#pragma endregion
+#endif
+
+/////////////////////////////// DD Components //////////////////////////////////////
+#ifndef __INTEL_COMPILER
+#pragma region DD Components
+#endif
+class DDTeamCollection;
+
+// send request/signal to TeamCollection through interface
+// call synchronous method from components outside DDTeamCollection
+struct ITeamCollection {
+	struct Interface {
+		PromiseStream<GetTeamRequest> getTeam;
+	};
+	virtual ~ITeamCollection() {}
+};
+
+// send request/signal to DDTracker through interface
+// call synchronous method from components outside DDTracker
+struct IDDTracker {
```

Comment:
I think it would be good to rename this IDDShardTracker. There is scope for confusion with TeamTracker otherwise.